### PR TITLE
feat(postcards): initial filler post endpoint

### PR DESCRIPTION
**Pull request by soph-iest**
**What:** Full frame + basic testing for forthcoming POST endpoint

 **Why:** https://www.notion.so/lob/Service-Tests-d853279999cd48ce8051f9522e88c782

**Details:** Mostly just configuring a lot of angular-specific stuff for testing.

### CODE REVIEW

File: test/services/postcard.spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      Sinon.stub(API, 'post').returns($q.resolve());
+
+      Postcard.create(payload);
+
+      $rootScope.$apply();
+
+      expect(API.post.firstCall.args[0]).to.eql(Config.API_HOST + '/postcards');
+      expect(API.post.firstCall.args[1]).to.eql(payload);
+
+      API.post.restore();
```
Comment (sjl2) [inline test/services/postcard.spec.js]: Not sure what version of `Sinon` we're using here, but in the latest version, you can globally restore all stubs so you don't need to do it in tests. It's p neat. No need to change here.

File: app/services/postcard.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+var Config = require('../../config');
+
+module.exports = function (API) {
+
+  var Postcard = {};
+
+  Postcard.url = Config.API_HOST + '/postcards';
```
Comment (sjl2) [inline app/services/postcard.js]: super nit: Not sure if it's possible with front-end, but the preferred way to concatenate strings with via interpolation:
```js
Postcard.url = `${Config.API_HOST}/postcards`
```

File: test/services/postcard.spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    it('calls the correct endpoint and with correct params', function () {
+      var payload = { id: 'psc_id' };
+
+      Sinon.stub(API, 'post').returns($q.resolve());
+
+      Postcard.create(payload);
+
+      $rootScope.$apply();
+
+      expect(API.post.firstCall.args[0]).to.eql(Config.API_HOST + '/postcards');
```
Comment (sjl2) [inline test/services/postcard.spec.js]: Again, not sure what version of Sinon we provide, but in most of our code bases, we import `sinon-chai`, so you will be able to use:

```
expect(API.post).to.have.been.calledOnceWith(url, payload);
```

File: test/services/postcard.spec.js (hunk 1 above)
Comment (soph-iest) [reply test/services/postcard.spec.js]: Nice!

File: app/services/postcard.js (hunk 2 above)
Comment (soph-iest) [reply app/services/postcard.js]: It totally is, I just didn't correct that in the copy-pasted code. Should I update this in the tutorial?

File: test/services/postcard.spec.js (hunk 3 above)
Comment (soph-iest) [reply test/services/postcard.spec.js]: Oh, nice, that's much cleaner. I'll probably maintain the version of Sinon in this repo for now.

File: app/services/postcard.js (hunk 2 above)
Comment (itsachen) [reply app/services/postcard.js]: RIP - is the angular tutorial not ES6 compatible?

File: test/services/postcard.spec.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+var expect  = require('chai').expect;
+var Angular = require('angular');
+var Sinon   = require('sinon');
+
+var Config = require('../../config');
+
+require('angular-mocks');
+
+
```
Comment (itsachen) [inline test/services/postcard.spec.js]: Syntax suggestion: This extra newline may not be necessary

File: app/services/postcard.js (hunk 2 above)
Comment (soph-iest) [reply app/services/postcard.js]: Updated the configurations, all ES6 now!

**Pull request merged**